main.cpp

```diff
@@ -105,10 +101,10 @@ struct MyUDT2
     {
         std::cout << "U's oldFunk value: " << that->oldFunk << std::endl;
```

Comment:
FIXME: FIXME: you're still using `that` without checking if it's a nullptr first!
- [ ] revise